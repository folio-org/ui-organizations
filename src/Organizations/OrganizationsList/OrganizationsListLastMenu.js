import React from 'react';
import { FormattedMessage } from 'react-intl';
import { useLocation } from 'react-router-dom';

import {
  PaneMenu,
  Button,
} from '@folio/stripes/components';
import { IfPermission } from '@folio/stripes/core';

const OrganizationsListLastMenu = () => {
  const { search } = useLocation();

  return (
    <IfPermission perm="ui-organizations.create">
      <PaneMenu>
        <FormattedMessage id="stripes-smart-components.addNew">
          {ariaLabel => (
            <Button
              id="clickable-neworganization"
              aria-label={ariaLabel}
              to={{
                pathname: '/organizations/create',
                search,
              }}
              buttonStyle="primary"
              marginBottom0
            >
              <FormattedMessage id="stripes-smart-components.new" />
            </Button>
          )}
        </FormattedMessage>
      </PaneMenu>
    </IfPermission>
  );
};

export default OrganizationsListLastMenu;
