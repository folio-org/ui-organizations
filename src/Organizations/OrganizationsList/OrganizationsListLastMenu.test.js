import React from 'react';
import { render } from '@folio/jest-config-stripes/testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import OrganizationsListLastMenu from './OrganizationsListLastMenu';

const renderOrganizationsListLastMenu = () => render(
  <OrganizationsListLastMenu />,
  { wrapper: MemoryRouter },
);

describe('OrganizationsListLastMenu', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct structure', async () => {
    const { asFragment } = renderOrganizationsListLastMenu();

    expect(asFragment()).toMatchSnapshot();
  });
});
