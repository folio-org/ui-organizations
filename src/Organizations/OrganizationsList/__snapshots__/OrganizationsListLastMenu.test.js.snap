// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationsListLastMenu should render correct structure 1`] = `
<DocumentFragment>
  <div
    class="paneMenu"
  >
    <a
      aria-label="stripes-smart-components.addNew"
      class="button primary marginBottom0"
      href="/organizations/create"
      id="clickable-neworganization"
      role="button"
    >
      stripes-smart-components.new
    </a>
  </div>
</DocumentFragment>
`;
